kong/db/dao/wasm_filter_chains.lua

```diff
@@ -0,0 +1,108 @@
+local wasm_filter_chains = {}
+
+local insert = table.insert
+local fmt = string.format
+
+local EMPTY = {}
+
+
+local function check_enabled_filters(self, chain)
+  if not self.filters then
+    local err_t = self.errors:schema_violation({
+      filters = "no wasm filters are configured",
+    })
+    return nil, tostring(err_t), err_t
+  end
+
+  if type(chain.filters) ~= "table" then
+    return true
+  end
+
+  local errs
+
+  for i, filter in ipairs(chain.filters) do
+    local name = filter.name
+
+    -- let the standard schema validation catch invalid name errors
+    if type(name) == "string"
+       and not self.filters_by_name[name]
+    then
+      errs = errs or {}
+      errs[i] = { name = "no such filter: " .. filter.name }
```

Comment:
what about

```lua
    local err = {
      name = "no such filter: " .. name
    }
    errs = errs or {}
    table.insert(errs, err)
```

Reply:
Using the explicit index instead of `table.insert()` does look odd as it will create a sparse table, but this is done in order to maintain parity with the way our schema validation errors are formatted (the index within the error should match the index from the user input).